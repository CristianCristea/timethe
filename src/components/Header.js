import React from "react";
import PropTypes from "prop-types";
import { Container, Row, Col, Jumbotron } from "reactstrap";

export default function Header({ title, subtitle }) {
  return (
    <header>
      <Jumbotron>
        <Container>
          <Row>
            <Col>
              <h1 className="display-3">{title}</h1>
              <p className="lead">{subtitle}</p>
            </Col>
          </Row>
        </Container>
      </Jumbotron>
    </header>
  );
}

Header.propTypes = {
  title: PropTypes.string.isRequired
};
