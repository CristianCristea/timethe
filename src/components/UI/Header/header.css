.logo {
  color: inherit; }
  .logo:hover {
    text-decoration: none;
    color: inherit; }
