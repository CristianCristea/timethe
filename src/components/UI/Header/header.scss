// import variables - colors;
@import '../../../scss/config';

.logo {
  color: inherit;

  &:hover {
    text-decoration: none;
    color: inherit;
  }
}

.hero {
  background-color: $primary-color;
  color: #fff;
  border-radius: 0;
}
