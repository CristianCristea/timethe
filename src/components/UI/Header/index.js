import React from "react";
import PropTypes from "prop-types";
import { Container, Row, Col, Jumbotron } from "reactstrap";
import "./header.css";
import { Link } from "react-router-dom";

export default function Header({ title, subtitle }) {
  return (
    <header>
      <Jumbotron>
        <Container>
          <Row>
            <Col>
              <Link to="/" className="display-3 m-0 logo">
                {title}
              </Link>
              <p className="lead">{subtitle}</p>
            </Col>
          </Row>
        </Container>
      </Jumbotron>
    </header>
  );
}

Header.propTypes = {
  title: PropTypes.string.isRequired
};
