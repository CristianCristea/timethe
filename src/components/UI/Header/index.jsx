import React from 'react';
import PropTypes from 'prop-types';
import { Container, Row, Col, Jumbotron } from 'reactstrap';
import { Link } from 'react-router-dom';
import './header.css';

export default function Header({ title, subtitle }) {
  return (
    <header>
      <Jumbotron>
        <Container>
          <Row>
            <Col>
              <Link to="/" className="display-3 m-0 logo">
                {title}
              </Link>
              <p className="lead">{subtitle}</p>
            </Col>
          </Row>
        </Container>
      </Jumbotron>
    </header>
  );
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};
