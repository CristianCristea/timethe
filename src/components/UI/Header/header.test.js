/* global it, expect */
import React from 'react';
import { shallow } from 'enzyme';
import Header from './index';

it('should render header', () => {
  const wrapper = shallow(<Header title='TIMETHE' subtitle='test' />);
  expect(wrapper).toMatchSnapshot();
});
