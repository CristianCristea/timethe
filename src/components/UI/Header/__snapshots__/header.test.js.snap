// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header 1`] = `
<header>
  <Jumbotron
    className="m-0 hero"
    tag="div"
  >
    <Container
      tag="div"
    >
      <Row
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Link
            className="display-3 m-0 logo"
            replace={false}
            to="/dashboard"
          >
            TIMETHE
          </Link>
          <p
            className="lead"
          >
            test
          </p>
        </Col>
      </Row>
    </Container>
  </Jumbotron>
</header>
`;
